projects/common/src/instrumentation-quality/instrumentation-quality.service.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpClient } from '@angular/common/http';
+import { Inject, Injectable, InjectionToken } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+import { InMemoryStorage } from '../utilities/browser/storage/in-memory-storage';
+
+export const INSTRUMENTATION_QUALITY_OPTIONS = new InjectionToken<TokenOptions>('INSTRUMENTATION_QUALITY_OPTIONS');
+
+@Injectable({
+  providedIn: 'root'
+})
+export class InstrumentationQualityService {
+  public BASE_URL: string;
+  public ORG_STORAGE_KEY: string = 'qoi-org-score';
+
+  public constructor(
+    private readonly http: HttpClient,
+    private readonly inMemoryStorage: InMemoryStorage,
+    @Inject(INSTRUMENTATION_QUALITY_OPTIONS) tokenOptions: TokenOptions
+  ) {
+    this.BASE_URL = `${tokenOptions.uri}/v1/score`;
+  }
+
+  public get<T>(endPoint: string = ''): Observable<T> {
+    return this.http.get<T>(this.BASE_URL + endPoint);
+  }
```

Comment:
Don't see an advantage of having this here. If you're making one API call, this seems more like an overhead here.

Reply:
There are 2 API calls:
1. service score
2. org score
And another one for BU scores will be added in future. All share the same BASE_URL. 

Need to move the whole token injection code for getting BASE_URL to the [component service][1] if this is removed. Should I move it?

[1]: https://github.com/razorpay/hypertrace-ui/blob/qoi-details/projects/observability/src/pages/apis/service-detail/instrumentation/service-instrumentation.service.ts#L10

Reply:
Or maybe the name `get` is confusing? There are 2 methods in this file. First one's used to make the service score API call, second one makes the org score API call. I can rename it to `getServiceScore`.

Reply:
I find the presence of `get` to be an overhead. I feel we can have two dedicated methods for each respective call.

Reply:
`BASE_URL` can be `global` in the file. Each method can have their respective `endpoint` defined. Imagine if any call requires any specific header or something specific in the future. For `user-preference`, we did it as we were sending the `user-email` header. Here, I don't think it's needed. Token injection is not a problem. Having a reference to a method which just `get`s the response is overhead for me.

Reply:
The first method is the dedicated one for getting service score. The `endpoint` is just the name of the service, maybe the naming caused the confusion.

If we just call the BASE_URL without adding anything, we get the org score:
https://qoi-qs.concierge.stage.razorpay.in/v1/score/

Add the service name, we'll get the service score:
https://qoi-qs.concierge.stage.razorpay.in/v1/score/cache

✅ Refactored the method to this:
```ts
  public getServiceScore<T>(serviceName: string): Observable<T> {
    return this.http.get<T>(this.BASE_URL + serviceName);
  }
```